haskell/lint.bzl

```diff
@@ -53,14 +59,31 @@ def _haskell_lint_aspect_impl(target, ctx):
         "--make",
     ])
 
-    args.add(expose_packages(
-        build_info,
-        lib_info,
-        use_direct = False,
-        use_my_pkg_id = None,
-        custom_package_caches = None,
-        version = ctx.rule.attr.version,
-    ))
+    args.add(lib_info.ghc_args if lib_info != None else bin_info.ghc_args)
+
+    # External libraries.
```

Comment:
This section of code looks like a duplicate of code in `link.bzl`. Any chance this can be factored out?

Reply:
Parts of this code seems to be duplicated in several places through the codebase (like [here](https://github.com/tweag/rules_haskell/blob/master/haskell/private/actions/link.bzl#L222-L230), [here](https://github.com/tweag/rules_haskell/blob/master/haskell/private/actions/repl.bzl#L60-L65) or [here](https://github.com/tweag/rules_haskell/blob/master/haskell/doctest.bzl#L112-L126)), I'd like to take some time to understand what can be factored out and how − possibly taking nixpkgs as a reference since they manage to get most of the tooling to work out-of-the box without having to pass a complex command-line to each of them.

Reply:
I added an issue about the copypaste: https://github.com/tweag/rules_haskell/issues/690

Doesn’t mean we should add new duplicates though. :P